### PR TITLE
fix: preserve repeated active progress events

### PR DESCRIPTION
**Pull request by hawkingrei**
## Summary

- Preserve active TUI progress events as an ordered event stream instead of deduplicating repeated messages.
- Keep repeated `Exploring`, `Planning`, and `Running` entries visible so long-running turns render like Codex-style appended history cells.
- Add a focused regression test for consecutive duplicate progress events.

## Why

The current active progress state drops repeated adjacent events with the same role/message. In long turns this makes `Exploring` / `Thinking` / `Ran` style output look like it is updating the same area instead of appending the actual event sequence.

## Testing

- `cargo test active_turn_cell_preserves_consecutive_duplicate_progress_events -- --nocapture`
- `cargo test active_turn_cell -- --nocapture`
- `cargo check`
- `git diff --check`

### CODE REVIEW

File: src/tui/state/transcript.rs [hunk 1]
```diff
@@ -324,79 +324,36 @@ impl TuiApp {
...
-        if self
-            .active_live
-            .events
-            .last()
-            .is_some_and(|last| last.role() == event.role() && last.message() == event.message())
-        {
-            return;
-        }
         self.active_live.events.push(event);
     }
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply src/tui/state/transcript.rs:328]: Fixed: ordered_exploration_agent_segments no longer deduplicates restored/materialized exploration entries, and there is regression coverage for duplicate restored exploration segments.

File: src/tui/state/transcript.rs [hunk 2]
```diff
@@ -324,79 +324,36 @@ impl TuiApp {
...
-            .active_live
-            .exploration_actions
-            .iter()
-            .any(|item| item == &action)
-        {
-            self.active_live.exploration_actions.push(action.clone());
-        }
+        self.active_live.exploration_actions.push(action.clone());
         self.push_active_live_event(ActiveLiveEvent::ExplorationAction(action));
     }
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply src/tui/state/transcript.rs:341]: Fixed: active_live.events remains the full ordered event stream, while the legacy per-section vectors are deduped again so they stay compact summary state.
